src/apolo_app_types/protocols/common/k8s.py

```diff
@@ -54,33 +72,71 @@ class Container(AbstractAppFieldType):
     model_config = ConfigDict(
         protected_namespaces=(),
         json_schema_extra=SchemaExtraMetadata(
-            title="Container",
-            description="K8S container configuration.",
+            title="Container Configuration",
+            description="Define command, arguments,"
+            " and environment variables for the"
+            " Kubernetes container.",
+        ).as_json_schema_extra(),
+    )
+
+    command: list[str] | None = Field(
+        default=None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Container Command",
+            description="Override the container's default "
+            "entrypoint by specifying a custom command.",
+        ).as_json_schema_extra(),
+    )
+
+    args: list[str] | None = Field(
+        default=None,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Container Arguments",
+            description="Provide arguments to pass to the "
+            "container's entrypoint or command.",
+        ).as_json_schema_extra(),
+    )
+
+    env: list[Env] = Field(
+        default_factory=list,
+        json_schema_extra=SchemaExtraMetadata(
+            title="Environment Variables",
+            description="Define environment variables to inject into the container.",
         ).as_json_schema_extra(),
     )
-    command: list[str] | None = None
-    args: list[str] | None = None
-    env: list[Env] = Field(default_factory=list)
 
 
 class Port(BaseModel):
     name: str = Field(
         default="http",
-        title="Port Name",
-        description="Name of the port.",
+        json_schema_extra=SchemaExtraMetadata(
+            title="Port Name",
+            description="Specify a name for the port "
+            "(e.g., 'http', 'grpc') to identify it in the service.",
+        ).as_json_schema_extra(),
     )
+
     port: int = Field(
         default=80,
-        title="Port",
-        description="Port number.",
+        json_schema_extra=SchemaExtraMetadata(
+            title="Port Number",
+            description="Set the port number to expose from the container.",
+        ).as_json_schema_extra(),
     )
+
     path_type: str = Field(
         default="Prefix",
-        title="PathType",
-        description="Type of path",
+        json_schema_extra=SchemaExtraMetadata(
+            title="Path Type",
+            description="Define how the path should be matched "
+            "(e.g., 'Prefix' or 'Exact').",
+        ).as_json_schema_extra(),
     )
+
     path: str = Field(
         default="/",
-        title="Path",
-        description="Path for the port.",
+        json_schema_extra=SchemaExtraMetadata(
```

Comment:
nit, maybe we should define own field type :thinking:

Reply:
in this way, we could validate model definitions via metaclass to enforce it's usage, with titles, descriptions ,etc